src/SourceBuild/content/eng/pipelines/ci.yml

```diff
@@ -16,6 +16,19 @@ pr:
     - internal/release/*
 
 stages:
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+  - template: templates/stages/vmr-scan.yml
```

Comment:
The use of two new stages is concerning to me from the perspective of build agent usage.  The time taken to allocate an agent is going to take far longer than the time to run the actual stage.  Given the constraints we face on build agents this doesn't feel like good usage of our resources.  Can these two stages be combined into one?

Reply:
I made this run on the `ubuntu-latest` AzDO agents which are instant so the whole run takes 2 minutes E2E. I will check whether we're allowed to use these agents for this purpose.

Reply:
Instant?  The build agents still have to be provisioned don't they?  If the load is heavy would it be instant? 

cc @MattGal for his input.

Reply:
They are pre-spun AzDO agents coming from the Azure Hosted pool, not the 1ES pool of VMs that provisions on demand. 

We are only scanning a public repo with this, making no writes or changes so I am inclined to hope we can use these. But I need to check the policy. There will be internal code in the cloned VMR later though..

But yeah, the clone+toolset init is running twice (it only takes a minute though). I can merge those, I have no strong preference, I only split them so the build summary would clearly show which one of these is alerting.

Reply:
This should be scanning the dnceng mirrors of the public repo since it is running in internal CI.  I do not believe it should not be using the hosted agents.

Reply:
Yeah, I realized that as I sent the last comment. I will make them 1ES machines and merge into a single job.

Reply:
Done